### ISSUE TITLE
Using osc with yarn + webpack fails on build.

### ISSUE THREAD

**Issue by s-light** [colinbdclark/osc.js#86]
I successfully used osc.js in an 'vanilla' javascript [tool/lib](https://github.com/s-light/tuio.js).
now i tried to use the same lib in combination with yarn and webpack.
i installed with  `yarn add osc`
and tried to build with `./node_modules/webpack/bin/webpack.js --progress`
this gave me a huge list of warnings and errors -  the first error is
```
ERROR in ./~/osc/src/platforms/osc-node.js
Module not found: Error: Can't resolve 'dgram' in '/home/stefan/xxx/github/tuio.js-example-webpack/node_modules/osc/src/platforms'
```
the complete log can be found at https://gist.github.com/s-light/e4f20c90f0e6883a55e5460b1ed6e4b4
my test repository is at https://github.com/s-light/tuio.js-example-webpack

i think that someway i have not setup this correctly. because i only want to use the osc in browser. (no server side things.. and as fare as i found dgram is a node.js thing..)
can someone point me to what i have missed?

**Comment by colinbdclark** [comment]
If you just want to use osc.js in the browser, you can  use the pre-built browser file located in the <code>dist/</code> directory of the osc.js module you installed with yarn, [as described in the osc.js README](https://github.com/colinbdclark/osc.js#including-oscjs-in-your-html-page).

I don't know of any reason why you'd need (or want) to run webpack on osc.js' Node.js source code.

**Comment by s-light** [comment]
i don't want to run webpack on the osc.js Node.js code...
i would like to run webpack on the source that yields the `osc-browser.js` file...

the pre-built browser file worked as described in the readme with 'vanilla javascript'.
iam new to the 'webpack & yarn - world' - as fare as i have understand the concept - the idea is to have only on bundled/generated file from webpack that is include in the html code. (bundle.js or similar).
during this bundling webpack also can help so that imported libraries are only bundle ones - also they are imported by several modules...

---
short: using the prebuild file in an script tag and add
```
module.exports = {
    // ...
    externals: {
        osc: 'osc'
    },
    // ...
}
```
to the `webpack.config.js` worked for me.
with this i can use it 'normally' with an `import osc from 'osc'` statement.

---
TL;DR;
to learn and understand better i experimented with different things / tips i found across the web to related problems:

1. i tried with only coping `osc-browser.js` file into my src directory and removed the dependency with `yarn remove osc` and import it with `import osc from './osc-browser`
then the bundling also generates errors:
```
ERROR in ./src/osc-browser.js
Module not found: Error: Can't resolve './osc.js' in '/home/stefan/xxx/github/tuio.js-example-webpack/src'
 @ ./src/osc-browser.js 2961:17-36
 @ ./src/index.js

ERROR in ./src/osc-browser.js
Module not found: Error: Can't resolve '../osc.js' in '/home/stefan/xxx/github/tuio.js-example-webpack/src'
 @ ./src/osc-browser.js 3182:17-37
 @ ./src/index.js

ERROR in ./src/osc-browser.js
Module not found: Error: Can't resolve 'long' in '/home/stefan/xxx/github/tuio.js-example-webpack/src'
 @ ./src/osc-browser.js 57:21-36
 @ ./src/index.js

ERROR in ./src/osc-browser.js
Module not found: Error: Can't resolve 'slip' in '/home/stefan/xxx/github/tuio.js-example-webpack/src'
 @ ./src/osc-browser.js 2962:19-34
 @ ./src/index.js

ERROR in ./src/osc-browser.js
Module not found: Error: Can't resolve 'ws' in '/home/stefan/xxx/github/tuio.js-example-webpack/src'
 @ ./src/osc-browser.js 3188:67-81
 @ ./src/index.js
```
they all seems valid. 
webpack scans the file and tries to resolve the imports... the files are not there ;-) but i don't know why the osc-browser.js file imports these.

2. for this i found if i add `module: { noParse: /osc-browser/ }` to my `webpack.config.js` it does not try to pares it so the error message vanish. 
full webpack config as reference how to use these:
`webpack.config.js`
```
var path = require('path');

module.exports = {
    devtool: "source-map",
    entry: './src/index.js',
    module: {
        noParse: /osc-browser/,
    },
    output: {
        filename: 'bundle.js',
        path: path.resolve(__dirname, 'dist')
    }
};
```
but if i try it this way i get an 
`11:06:21.089 ReferenceError: require is not defined osc-browser.js:2962`
the [documentation for the noParse](https://webpack.js.org/configuration/module/#module-noparse) says:

> Ignored files should not have calls to import, require, define or any other importing mechanism.

3. i also tried to alias the file directly so i don't need to copy it. for this i tried to use [resolve.alias](https://webpack.js.org/configuration/resolve/#resolve-alias) as following:
```
module.exports = {
    // ...
    resolve: {
        alias: { osc$: "osc/dist/osc-browser.js" }
    },
    // ...
}
```
with this i got similar errors:
```
$ ./node_modules/webpack/bin/webpack.js --progress
Hash: 73cee5637601b5121136
Version: webpack 2.3.1
Time: 3745ms
        Asset     Size  Chunks                    Chunk Names
    bundle.js  1.23 MB       0  [emitted]  [big]  main
bundle.js.map  1.49 MB       0  [emitted]         main
   [0] ./~/buffer/index.js 48.6 kB {0} [built]
   [3] ./~/process/browser.js 5.3 kB {0} [built]
   [7] ./~/events/events.js 8.33 kB {0} [built]
  [59] ./~/long/dist/long.js 39.8 kB {0} [built]
  [82] ./~/ws/lib/WebSocket.js 27.6 kB {0} [built]
  [83] ./src/tuio.js 20.8 kB {0} [built]
  [93] ./~/base64-js/index.js 3.48 kB {0} [built]
 [137] ./~/ieee754/index.js 2.05 kB {0} [built]
 [139] ./~/osc/dist/osc-browser.js 108 kB {0} [built]
 [140] ./~/osc/dist/osc.js 34.9 kB {0} [built]
 [171] ./~/slip/src/slip.js 5.84 kB {0} [built]
 [182] ./~/wolfy87-eventemitter/EventEmitter.js 18.8 kB {0} [built]
 [183] ./~/ws/index.js 1.05 kB {0} [built]
 [190] ./~/ws/lib/WebSocketServer.js 16.9 kB {0} [built]
 [192] ./src/index.js 1.94 kB {0} [built]
    + 180 hidden modules

WARNING in ./~/ws/lib/BufferUtil.js
Module not found: Error: Can't resolve 'bufferutil' in '/home/stefan/_LocalData/projects/github/tuio.js-example-webpack/node_modules/ws/lib'
 @ ./~/ws/lib/BufferUtil.js 10:19-40
 @ ./~/ws/lib/Receiver.js
 @ ./~/ws/index.js
 @ ./~/osc/dist/osc-browser.js
 @ ./src/tuio.js
 @ ./src/index.js

WARNING in ./~/ws/lib/Validation.js
Module not found: Error: Can't resolve 'utf-8-validate' in '/home/stefan/_LocalData/projects/github/tuio.js-example-webpack/node_modules/ws/lib'
 @ ./~/ws/lib/Validation.js 10:19-44
 @ ./~/ws/lib/Receiver.js
 @ ./~/ws/index.js
 @ ./~/osc/dist/osc-browser.js
 @ ./src/tuio.js
 @ ./src/index.js

ERROR in ./~/osc/dist/osc-browser.js
Module not found: Error: Can't resolve '../osc.js' in '/home/stefan/_LocalData/projects/github/tuio.js-example-webpack/node_modules/osc/dist'
 @ ./~/osc/dist/osc-browser.js 3182:17-37
 @ ./src/tuio.js
 @ ./src/index.js

ERROR in ./~/ws/lib/WebSocketServer.js
Module not found: Error: Can't resolve 'tls' in '/home/stefan/_LocalData/projects/github/tuio.js-example-webpack/node_modules/ws/lib'
 @ ./~/ws/lib/WebSocketServer.js 15:10-24
 @ ./~/ws/index.js
 @ ./~/osc/dist/osc-browser.js
 @ ./src/tuio.js
 @ ./src/index.js

ERROR in ./~/options/lib/options.js
Module not found: Error: Can't resolve 'fs' in '/home/stefan/_LocalData/projects/github/tuio.js-example-webpack/node_modules/options/lib'
 @ ./~/options/lib/options.js 6:9-22
 @ ./~/ws/lib/WebSocket.js
 @ ./~/ws/index.js
 @ ./~/osc/dist/osc-browser.js
 @ ./src/tuio.js
 @ ./src/index.js
```
if combine this with the noParse option i get the same behavior as with the copied `osc-browser.js` with noParse `require is not defined` error in browser console.

had no more ideas what to try... for now i will use the scripttag + external variant mentioned at the beginning.

**Comment by breedx2** [comment]
For others that may stumble upon this and want to webpack osc into their client-side application (without having yet another `<script>` tag), I was able to this this working pretty simply by adding `osc` to my webpack resolve alias:
```
resolve: {
    alias: {
        'osc': path.join(__dirname, 'node_modules/osc/src/osc.js'),
        ...
    }
}
```

**Comment by colinbdclark** [comment]
Thanks, @breedx2, that's awesome! I really appreciate the information, and am very glad to hear that osc.js is  interoperable with webpack.

I'd love to turn this into some documentation in the main osc.js README, but not being a webpack user, I might need some help from others to elaborate what's required on the part of a webpack user.
